src/main/java/tc/oc/pgm/commands/ModerationCommands.java

```diff
@@ -0,0 +1,70 @@
+package tc.oc.pgm.commands;
+
+import app.ashcon.intake.Command;
+import app.ashcon.intake.parametric.annotation.Text;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.Bukkit;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import tc.oc.component.Component;
+import tc.oc.component.types.PersonalizedText;
+import tc.oc.component.types.PersonalizedTranslatable;
+import tc.oc.named.NameStyle;
+import tc.oc.pgm.api.Permissions;
+import tc.oc.pgm.api.chat.Audience;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.events.PlayerReportEvent;
+
+public class ModerationCommands {
+
+  @Command(
+      aliases = {"report"},
+      usage = "<player> <reason>",
+      desc = "Report a player who is breaking the rules")
+  public static void report(
+      CommandSender commandSender,
+      MatchPlayer matchPlayer,
+      Match match,
+      Player player,
+      @Text String reason) {
+    MatchPlayer accused = match.getPlayer(player);
```

Comment:
What will happen if the player is in another match? i don't think the Match should be a parameter for this command, you should make use of the MatchManager to get the MatchPlayer

Reply:
I think the report should be scoped to the Match. The issue here is you need to check that `accused != null`

Reply:
> What will happen if the player is in another match?

We don't have the ability to run multiple matches at the moment and I don't think every new change needs to unnecessary pander to this idea. 

Regardless you shouldn't be reporting players from alternative matches because as you said yourself they should act independently. The audiences that it broadcast the notification message to could be server-wide as moderators may wish to switch match to investigate.

> you need to check that `accused != null`

When would the accused not be a player? The command framework seems pretty hell-bent on making sure the correct arguments are passed to the function.

![image](https://user-images.githubusercontent.com/8608892/71772520-31eb3080-2f44-11ea-92ca-ecc0699578d1.png)